src/webxdc/integration.rs

```diff
@@ -146,4 +149,65 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_overwrite_default_integration() -> Result<()> {
+        let t = TestContext::new_alice().await;
+        let self_chat = &t.get_self_chat().await;
+        assert!(t.init_webxdc_integration(None).await?.is_none());
+
+        async fn assert_integration(t: &Context, name: &str) -> Result<()> {
+            let integration_id = t.init_webxdc_integration(None).await?.unwrap();
+            let integration = Message::load_from_db(t, integration_id).await?;
+            let integration_info = integration.get_webxdc_info(t).await?;
+            assert_eq!(integration_info.name, name);
+            Ok(())
+        }
+
+        // set default integration
+        let bytes = include_bytes!("../../test-data/webxdc/with-manifest-and-png-icon.xdc");
+        let file = t.get_blobdir().join("maps.xdc");
+        tokio::fs::write(&file, bytes).await.unwrap();
+        t.set_webxdc_integration(file.to_str().unwrap()).await?;
+        assert_integration(&t, "with some icon").await?;
+
+        // send a maps.xdc with insufficient manifest
+        let mut msg = Message::new(Viewtype::Webxdc);
+        msg.set_file_from_bytes(
+            &t,
+            "mapstest.xdc",
+            include_bytes!("../../test-data/webxdc/mapstest-integration-unset.xdc"),
+            None,
+        )
+        .await?;
+        chat::send_msg(&t, self_chat.id, &mut msg).await?;
+        let sent = t.pop_sent_msg_opt(Duration::from_secs(1)).await;
+        assert!(sent.is_some());
+        assert_integration(&t, "with some icon").await?; // still the default integration
+
+        // send a maps.xdc with manifest including the line `integration = "maps"`
+        let mut msg = Message::new(Viewtype::Webxdc);
+        msg.set_file_from_bytes(
+            &t,
+            "mapstest.xdc",
+            include_bytes!("../../test-data/webxdc/mapstest-integration-set.xdc"),
+            None,
+        )
+        .await?;
+        chat::send_msg(&t, self_chat.id, &mut msg).await?;
+        let sent = t.pop_sent_msg_opt(Duration::from_secs(1)).await;
+        assert!(sent.is_some());
+        assert_integration(&t, "Maps Test 2").await?;
+
+        // when maps.xdc is received on another device, the integration is not accepted
```

Comment:
Should it be sent at all then? (i'm not about the test, but the core behaviour)
Also if the integration isn't accepted on another device, if you make a backup before replacing the integration, it won't be replaced after the restoration. Are there any ideas how to fix this?

Reply:
> Should it be sent at all then?

yes. otherwise, the UI element of "Saved Messages can be used to send things to other devices"  get lost/weakened.

also, for now, it is not wanted to not replace the map remotely on other devices. so that it is ignored by a backup is a wanted behaviour. there should be some explicit user action (which are only devs and maybe "power-user" currently)

> Also if the integration isn't accepted on another device, if you make a backup before replacing the integration, it won't be replaced after the restoration. Are there any ideas how to fix this?

no need to fix that, as said, it is wanted behaviour, at least in the current stage.

to re-apply the map, in practise, already now, the user can forward the message again to the SELF chat and it will be picked up.

(but i will change the summary line so that it gets clearer what really happend)